tests/data_management/test_stats4schools_smoking.py

```diff
@@ -0,0 +1,123 @@
+import pandas as pd
+import pytest
+from pandas.testing import assert_series_equal
+from template_project.data_management.stats4schools_smoking_template import (
+    _clean_gender,
+    _clean_highest_qualification,
+    _clean_marital_status,
+    _clean_smoke,
+    clean_smoking_stats4schools,
+)
+
+
+def assert_categorical_equal(left, right):
+    assert_series_equal(pd.Series(left), pd.Series(right))
+
+
+@pytest.fixture()
+def data():
+    data = {
+        "highest_qualification": [
+            "No Qualification",
+            "GCSE/CSE",
+            "GCSE/O Level",
+            "ONC/BTEC",
+            "Other/Sub Degree",
+            "Higher/Sub Degree",
+            "A Levels",
+            "Degree",
+        ],
+        "gender": 4 * ["Male"] + 4 * ["Female"],
+        "age": 4 * [20] + 4 * [20.5],
+        "marital_status": [
+            "Divorced",
+            "Widowed",
+            "Separated",
+            "Single",
+            "Married",
+            "Widowed",
+            "Separated",
+            "Married",
+        ],
+        "smoke": 4 * ["Yes"] + 4 * ["No"],
+    }
+    return pd.DataFrame(data)
+
+
+def test_age_is_integer(data):
+    got = clean_smoking_stats4schools(data)["age"]
+    exp = pd.Series(8 * [20], name="age")
+    assert_series_equal(got, exp)
+
+
+def test_smoke_numerical_is_numerical(data):
+    got = clean_smoking_stats4schools(data)["smoke_numerical"]
+    exp = pd.Series(4 * [1] + 4 * [0], name="smoke_numerical", dtype="int8")
+    assert_series_equal(got, exp)
+
+
+def test_clean_gender(data):
+    got = _clean_gender(data["gender"])
+    exp = pd.Categorical(
+        4 * ["Male"] + 4 * ["Female"],
+        categories=["Female", "Male"],
+        ordered=False,
+    )
+    assert_categorical_equal(got, exp)
+
+
+def test_clean_marital_status(data):
+    got = _clean_marital_status(data["marital_status"])
+    exp = pd.Categorical(
+        [
+            "Divorced",
+            "Widowed",
+            "Separated",
+            "Single",
+            "Married",
+            "Widowed",
+            "Separated",
+            "Married",
+        ],
+        categories=["Single", "Married", "Separated", "Divorced", "Widowed"],
+        ordered=False,
+    )
+    assert_categorical_equal(got, exp)
+
+
+def test_clean_smoke(data):
+    got = _clean_smoke(data["smoke"])
+    exp = pd.Categorical(
+        4 * ["Yes"] + 4 * ["No"],
+        categories=["No", "Yes"],
+        ordered=True,
+    )
+    assert_categorical_equal(got, exp)
+
+
+def test_clean_highest_qualification(data):
```

Comment:
Can you add a test where we pin down behaviour when there is a typo in a category? Just to show good practice. Not sure how often we should repeat.